gpu/features/test/test_normals.cpp

```diff
@@ -371,6 +371,59 @@ TEST(PCL_FeaturesGPU, normals_highlevel_4)
     }
 }
 
+TEST(PCL_FeaturesGPU, normal_estimation_1000_zeroes)
+{
+    // This number is magic, do not set to lower value.
+    // It may affect error reproducibility.
+    const std::size_t N = 1000;
+
+    const float radius_search = 2.0F;
+    const int max_results = 500;
+
+    PointCloud<PointXYZ>::Ptr cloud(new PointCloud<PointXYZ>());
+    cloud->points.assign(N, {0.0, 0.0, 0.0});
+
+    pcl::NormalEstimation<PointXYZ, Normal> ne;
+    ne.setInputCloud(cloud);
+    ne.setSearchMethod(pcl::search::KdTree<PointXYZ>::Ptr (new pcl::search::KdTree<PointXYZ>));
+    ne.setRadiusSearch(radius_search);
+
+    PointCloud<Normal>::Ptr normals(new PointCloud<Normal>());
+    ne.compute(*normals);
+
+    pcl::gpu::NormalEstimation::PointCloud cloud_device;
+    cloud_device.upload(cloud->points);
+
+    pcl::gpu::NormalEstimation ne_device;
+    ne_device.setInputCloud(cloud_device);
+    ne_device.setRadiusSearch(radius_search, max_results);
+
+    pcl::gpu::NormalEstimation::Normals normals_device;
+    ne_device.compute(normals_device);
+
+    std::vector<PointXYZ> downloaded;
+    normals_device.download(downloaded);
+
+    ASSERT_EQ(downloaded.size(), normals->points.size());
+
+    for(std::size_t i = 0; i < downloaded.size(); ++i)
+    {
+        Normal n = normals->points[i];
+
+        ASSERT_TRUE(std::isnan(n.normal_x));
```

Comment:
CPU part will pass.

Reply:
Maybe split the test in 2 smaller tests: check for the original crash, and check CPU output is same as GPU output. The magic number of points is required for only the first test right?

Reply:
> check for the original crash, and check CPU output is same as GPU output

Okay, split to three tests: CPU + NaNs, GPU + NaNs, GPU crash.